knimin/lib/data_access.py

```diff
@@ -2630,283 +2628,120 @@ def delete_study(self, study_id):
+                TRN.add(sql, sql_args, many=True)
+                TRN.execute()
+            if to_remove:
+                to_remove = tuple(to_remove)
+                sql = """DELETE FROM pm.study_sample
+                         WHERE study_id = %s AND sample_id IN %s"""
+                TRN.add(sql, [study_id, to_remove])
+                sql = "DELETE FROM pm.sample WHERE sample_id IN %s"
+                TRN.add(sql, [to_remove])
+                TRN.execute()
 
-    def read_samples(self, ids):
-        """Read properties and associated study IDs of samples, if they exist
+    def get_study_plated_samples(self, study_id):
+        """Gets the plated samples for the given study
 
         Parameters
         ----------
-        ids : list of str
-            IDs of the samples to read
+        study_id : int
+            The study id
 
         Returns
         -------
-        dict of dict
-            {sample_id : {is_blank : bool, barcode : str, notes : str,
-             study_ids : list of int}}
-            Properties of each sample: Whether this sample is blank, barcode,
-            and notes, and sorted study IDs associated with it
-
-        Notes
-        -----
-        If one or more sample IDs do not exist, the function will not raise an
-        error. Instead, these samples will be absent in the returning dict.
-        Therefore, this function combines existence checking and properties
-        retrieval in one command.
-        """
-        with TRN:
-            sql = """SELECT sample_id, is_blank, barcode, notes,
-                            array_agg(study_id ORDER BY study_id)
-                            AS study_ids
-                     FROM pm.sample
-                     JOIN pm.study_sample
-                     USING (sample_id)
-                     WHERE sample_id IN %s
-                     GROUP BY sample_id"""
-            TRN.add(sql, [tuple(ids)])
-            samples = {}
-            for row in TRN.execute_fetchindex():
-                samples[row[0]] = dict(row)
-                samples[row[0]].pop('sample_id')
-            return samples
-
-    def delete_samples(self, ids):
-        """Deletes existing samples along with their associations with studies
-
-        Parameters
-        ----------
-        ids : list of str
-            IDs of the samples to delete
+        dict {plate_id: samples}
+            The plated samples grouped by plate id
 
         Raises
         ------
         ValueError
-            If one or more sample IDs do not exist
-            If one or more samples are associated with one or more sample
-            plates, in which case, the sample plates have to be deleted
-            prior to the deletion of the samples
+            If the study ID does not exist
         """
         with TRN:
-            self._samples_exist(ids)
-            sql = """SELECT sample_plate_id, sample_id
+            self._study_exists(study_id)
+            sql = """SELECT sample_plate_id,
```

Comment:
That's correct. However, this function will not be used to render existing plates, but for checking which samples from the study have been already plated.

The function used to render the existing plates is `read_sample_plate_layout`